spec/lib/ansible/runner/credential/machine_credential_spec.rb

```diff
@@ -0,0 +1,103 @@
+require 'ansible/runner'
+require 'ansible/runner/credential'
+
+RSpec.describe Ansible::Runner::MachineCredential do
+  it ".auth_type is the correct Authentication sub-class" do
+    expect(described_class.auth_type).to eq("ManageIQ::Providers::EmbeddedAnsible::AutomationManager::MachineCredential")
+  end
+
+  context "with a credential object" do
+    around do |example|
+      Dir.mktmpdir("ansible-runner-credential-test") do |dir|
+        @base_dir = dir
+        example.run
+      end
+    end
+
+    let(:auth) do
+      auth_attributes = {
+        :userid            => "manageiq",
+        :password          => "secret",
+        :auth_key          => "key_data",
+        :become_username   => "root",
+        :become_password   => "othersecret",
+        :auth_key_password => "keypass",
+        :options           => {:become_method => "su"}
+      }
+      FactoryBot.create(:embedded_ansible_machine_credential, auth_attributes)
+    end
+
+    let(:cred) { described_class.new(auth.id, @base_dir) }
+
+    describe "#command_line" do
+      it "is correct with all attributes" do
+        expected = {
+          :become        => nil,
+          :become_user   => "root",
+          :become_method => "su",
+          :user          => "manageiq"
+        }
+        expect(cred.command_line).to eq(expected)
+      end
+
+      it "doesn't send :user if userid is not set" do
+        auth.update!(:userid => nil)
```

Comment:
This will only work if we don't reference `cred` before we make the authentication change because of the lazy evaluation of `let`.

I wasn't sure if this made the test too brittle for a slightly more DRY example. I can change this so we initialize the credential in every spec if this is too much magic.

Reply:
@NickLaMuro thoughts here?

Reply:
> I can change this so we initialize the credential in every spec if this is too much magic.

If you are going to be making other changes, I would say go ahead and do that (I think it would just be changing it to a `let!(:cred)` instead.  Possibly add a comment right above it saying "this is done on purpose since some specs rely on this being mutated.

Alternatively, for that spec specifically, you could just call `cred` right above it with a comment saying "this is done here intentionally".  Then all of the other specs can take advantage of the lazy eval, and the one case where it needs to be preloaded, we just do it in the spec manually.